joomla/joomla-cms
# Use a database iterator for exporting CSV records

Pull Request for Issue #22700

### Summary of Changes

This should make the CSV export a bit more memory efficient by using a database iterator for processing the database results versus trying to load the entire resultset into memory first.

A new method is added to the actionlogs model to fetch the results as a `JDatabaseIterator` versus an array of `stdClass` objects with the common code between the two `getLogs` methods moved into a private function to keep things DRY.

The helper method now checks for any valid iterable, which per the `is_iterable()` function added to PHP 7.1 is any array or object implementing the core `Traversable` interface and throws an Exception if a non-iterable is given.  The Symfony polyfill package for PHP 7.1 is pulled in to be able to use this function.

### Testing Instructions

Try to export the CSV file for action logs data.  Should still work on smaller data sets, should work better for larger data sets.

### Expected result

CSV file exports

### Actual result

Memory issues on larger data sets.
